src/commands/flinkStatements.ts

```diff
@@ -24,8 +43,140 @@ export async function viewStatementSqlCommand(statement: FlinkStatement): Promis
   await vscode.window.showTextDocument(doc, { preview: false });
 }
 
+/**
+  * Submit a Flink statement to a Flink cluster. Flow:
+  * Quickpick flow:
+	*  1) Chose flinksql, sql, or text document, preferring the current foreground editor.
+  *  2) Statement name (auto-generated from template pattern, but user can override)
+	  2) the Flink cluster to send to
+    3) also need to know at least the 'current database' (cluster name) to submit along with the catalog name (the env, infer-able from the chosen Flink cluster).
+  5) Submit!
+  6) Raise error box if any immediate submission errors.
+  7) Refresh the statements view if the view is set to include the cluster
+*/
+export async function submitFlinkStatementCommand(): Promise<void> {
+  // 1. Choose the document with the SQL to submit
+  const uriSchemes = ["file", "untitled"];
+  const languageIds = ["plaintext", "flinksql", "sql"];
+  const fileFilters = {
+    "FlinkSQL files": [".flinksql", ".sql"],
+  };
+  const statementBodyUri: vscode.Uri | undefined = await uriQuickpick(
+    uriSchemes,
+    languageIds,
+    fileFilters,
+  );
+  if (!statementBodyUri) {
+    return;
+  }
+
+  const document = await getEditorOrFileContents(statementBodyUri);
+  const statement = document.content;
+
+  // 2. Choose the statement name
+  const statementName = await determineFlinkStatementName();
+
+  // 3. Choose the Flink cluster to send to
+  const computePool = await flinkComputePoolQuickPick();
+  if (!computePool) {
+    logger.error("sumbitFlinkStatementCommand", "computePool is undefined");
+    return;
+  }
+
+  // 4. Choose the current / default database / aka kafka cluster
+  // within the environment.
+  const currentDatabaseKafkaCluster: KafkaCluster | undefined = await kafkaClusterQuickPick({
+    placeHolder: "Select the Kafka cluster to use as the default database for the statement",
+    filter: (cluster: KafkaCluster) => {
+      if (cluster.environmentId !== computePool.environmentId) {
+        return false;
+      }
+      // Any survivors matching environmentId check should be CCloudKafkaClusters,
+      // since the compute pool is ccloud.
+      const ccloudCluster = cluster as CCloudKafkaCluster;
+      return (
+        ccloudCluster.provider === computePool.provider &&
+        ccloudCluster.region === computePool.region
+      );
+    },
+  });
+  if (!currentDatabaseKafkaCluster) {
+    logger.error("sumbitFlinkStatementCommand", "currentDatabaseKafkaCluster is undefined");
+    return;
+  }
+  const currentDatabase = currentDatabaseKafkaCluster.name;
+
+  // 5. Prep to submit, submit.
+  const submission: IFlinkStatementSubmitParameters = {
+    statement,
+    statementName,
+    computePool,
+    properties: new FlinkSpecProperties({
+      currentDatabase,
+      currentCatalog: computePool.environmentId,
+      // TODO get at the user's local timezone
+      // localTimezone: "GMT-04:00",
```

Comment:
We need that?

Reply:
For parity with ccloud workspaces. It nice to have, https://github.com/confluentinc/vscode/issues/1579